File: sample-apps/java/src/test/java/com/example/AppTest.java
package com.example;

import org.junit.Test;
import static org.junit.Assert.*;

public class AppTest {

    @Test
    public void testGreetWithName() {
        App app = new App();
        assertEquals("Hello, DevOps!", app.greet("DevOps"));
    }

    @Test
    public void testGreetWithNull() {
        App app = new App();
        assertEquals("Hello, World!", app.greet(null));
    }

    @Test
    public void testGreetWithEmpty() {
        App app = new App();
        assertEquals("Hello, World!", app.greet(""));
    }

    @Test
    public void testAdd() {
        App app = new App();
        assertEquals(5, app.add(2, 3));
    }
}

File: sample-apps/php/tests/AppTest.php
<?php

use PHPUnit\Framework\TestCase;
use App\App;

class AppTest extends TestCase
{
    public function testGreetWithName(): void
    {
        $app = new App();
        $this->assertEquals('Hello, DevOps!', $app->greet('DevOps'));
    }

    public function testGreetWithNull(): void
    {
        $app = new App();
        $this->assertEquals('Hello, World!', $app->greet(null));
    }

    public function testGreetWithEmpty(): void
    {
        $app = new App();
        $this->assertEquals('Hello, World!', $app->greet(''));
    }

    public function testAdd(): void
    {
        $app = new App();
        $this->assertEquals(5, $app->add(2, 3));
    }
}
